Mathlib/CategoryTheory/Preadditive/SingleObj.lean

```diff
@@ -19,8 +19,8 @@ namespace CategoryTheory
 variable {α : Type _} [Ring α]
 
 instance : Preadditive (SingleObj α) where
-  add_comp _ _ _ f f' g := mul_add g f f'
-  comp_add _ _ _ f g g' := add_mul g g' f
+  add_comp _ _ _ f f' g := by exact mul_add g f f'
+  comp_add _ _ _ f g g' := by exact add_mul g g' f
```

Comment:
These changes are pretty weird, and I don't think they're justifiable outside the context of your other PR

Reply:
I don't disagree, but apparently the current proof is so fragile that it breaks when instance priorities change. Isn't that enough justification to change the current proof?
I'd like to get out of the state where proofs are as fragile as possible such that Lean accepts them, but the tiniest change breaks them. Of course, it is generally hard to find out whether you are in such a state.

Reply:
I'd claim that _every_ proof is to done extent fragile WRT changes to instance priorities: so this change is only relevant to the specific priority change you made.

For that reason, I think it should be in the same PR as those changes: otherwise someone might just undo it again in a cleanup pass.

The other viewpoint would be "use by exact everywhere", but if that's really the answer to avoiding fragility then that sounds like a lean bug...

The other changes in this PR are about being more explicit, which is good by itself: my objection to this one is that it's an elaboration hack that's isolated from the thing that justified it.

Reply:
Ok, that's fair.